### ISSUE TITLE
Adding fr-bepo support

### ISSUE THREAD

**Issue by raylemon** [microsoft/wslg#696]
# Environment
Windows 11 Win32NT             10.0.22000.0 Microsoft Windows NT 10.0.22000.0
WSL Arch Linux
WSLg 2

Run any app in wslg mode (like emacs) . Keyboard in WSLg still US (in non graphic mode, keyboard is recognized)

**WSL logs**: 

[10:57:25.383] convert_rdp_keyboard_to_xkb_rule_names: matching layout=(null) variant=(null) options=(null)
[10:57:25.383] rail_client_LanguageImeInfo_callback: Failed to switch to kbd_layout:0xb001040c kbd_type:0x4 kbd_subType:0x0

#  Expected behavior
Fr-bepo language must be recognized in wslg apps

**Comment by hideyukn88** [comment]
@raylemon, thanks for reporting the issue. If I understand correctly, French-BÉPO layout (https://en.wikipedia.org/wiki/B%C3%89PO) is not one of Windows Standard layout (https://docs.microsoft.com/en-us/globalization/windows-keyboard-layouts). Have you installed 3rd party keyboard layout driver? Thanks!

**Comment by raylemon** [comment]
Yes, I have 3rd party driver for Windows. All Linux distros have the fr-bepo layout.  If I add the layout in weston.ini, it’s working, but I need to redo the commands at every start.

**Comment by hideyukn88** [comment]
@raylemon, the problem here is that the remote desktop protocol informs the client keyboard layout to WSLg by keyboard layout ID, and the ID can't be consistent/unique in 3rd party keyboard layouts, that said it's possible there is other 3rd party French keyboard layout can have same ID as the layout you are using, but different layout, or other 3rd party fr-bepo layout has different ID. Thus for now, you will need to have workaround to set layout at every startup. This can be solved either by 1) Windows supports fr-bepo layout officially (so ID is assigned by Microsoft) or 2) WSLg supports the solution for https://github.com/microsoft/wslg/issues/173.

**Comment by bateast** [comment]
Hello,
now that 

1. Microsoft has announced the support of _bépo_ ([Microsoft blog](https://blogs.windows.com/windows-insider/2022/11/18/announcing-windows-11-insider-preview-build-25247/)), and upstream 
2. Weston source reference ([Weston rdp map](https://gitlab.freedesktop.org/wayland/weston/-/blob/main/libweston/backend-rdp/rdp.c#L863)) has been including _bépo_ for years ([RDP reference implementation](https://github.com/awakecoding/FreeRDP/blame/6d9d1185936318fb72f84bfec22d6e17aa3515c3/libfreerdp/locale/xkb_layout_ids.c#L303))

What and when can we expect ?
A weston update could be needed (_rdp map_ does not includes _bépo_ though its reference does)…

**Comment by hideyukn88** [comment]
@bateast, thanks for reminding us!

**Comment by bateast** [comment]
Hello,
thank you for the update.

Shouldn't it be bepo_afnor on line 882 ?
https://github.com/microsoft/weston-mirror/blob/70cbfad51df2b4cd62fd5a6f98f596a787250c66/libweston/backend-rdp/rdp.c#L881C24-L882C1

Especially given AltGr-Space (Alt-Space actually…) is inaccessible in WSL, underscore can not be used in deprecated bépo 1.0.
(https://github.com/microsoft/wslg/issues/1068 — actually a big deal for Linux apps…)

**Comment by CharlSka** [comment]
I finally found a workaround ! Not perfect because I have to run a script after each start or each layout change on Windows.

I found a script to run a portable version of my layout : https://cdn.bepo.fr/xkb/fr-dvorak-bepo-xkb-1.1rc2.tar.xz
(explanation in french are available here : https://bepo.fr/wiki/X.Org_portable)

If I try to change layout after running the script the switch dont work. It goes bach to Qwerty and I have to re-run the script.

I let the skilled people take a look to the script to understand :).
I hope this can help !
